bbc/simorgh
# WSTEAM1-953: Uzbek dual publishing routes

Resolves JIRA [WSTEAM1-953](https://jira.dev.bbc.co.uk/browse/WSTEAM1-953)

Overall changes
======
Update logic in Simorgh to convert Uzbek from single service to dual service

Code changes
======
- Theme Provider
-- Update Uzbek to use Cyr and Lat themes.
- Service Config: 
-- Split Uzbek serviceConfig into variants. (Cyr, Lat, Default) - note comments mark where translations are required
- Models/types/globals
-- Update Uzbek types to handle variants
- Other
-- Check and update areas in codebase that directly reference variants on services and update them to include Uzbek
-- Add/ update relevant unit tests
-- Add fixture data for Uzbek article in cyr and lat variants. Updates STY fixture data so it loads

Out of scope of this ticket:
- Variant Redirecting
-- Have not added logic here to redirect if you land on an article without a variant e.g. uzbek/articles/cxj3rjxm6r0o. We will look to remove/move our existing logic from Simorgh in a follow up ticket.
-- We have added some temporarily logic to the VariantHandler to prevent the homepage redirecting. This can be removed once the Uzbek Homepage defaults to /cyr
- Secondary data fetch
-- Will need to do a follow up task to check when/ whether this is possible to do

Some other things to note post swarm
- Secondary Data for Optimo Articles
-- topStories, features, mostRead and latestMedia are all coming through as null from the backend response so I have updated the fixture data to reflect this. (I think I had just copied over the original non-variant response, oops!)
- E2E Tests
-- Have added but not enabled for Uzbek articles cyr/ lat due to missing secondary data
-- Have enabled for Uzbek STY pages locally. (Could do for more enviornments and page types if we think it necessary).


Testing
======
1. Pull down this PR and run simorgh locally using `yarn dev`
2. Visit http://localhost:7080/uzbek/articles/cxj3rjxm6r0o/cyr and note headline. (Secondary data is missing)
3. Visit http://localhost:7080/uzbek/articles/cxj3rjxm6r0o/lat and note headline. (Secondary data is missing)
4. Visit http://localhost:7080/uzbek/sport-23248721 and check everything renders properly.

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
